File: ha1/src/main/java/Main.java
package main.java;

/**
 * Main class to demonstrate the functionality of the Circle, Rectangle, and Square classes.
 *
 * @author Simon Wessel
 * @version 1.3
 * @since 1.0
 */
public class Main {
    /**
     * Main method to demonstrate the functionality of the classes.
     *
     * @param args Command line arguments (not used).
     */
    public static void main(String[] args) {
        // Circles
        Circle circle1 = new Circle();
        Circle circle2 = new Circle(new Point(0,0), 5);
        Circle circle3 = Circle.fromArea(new Point(0,0), 50);

        System.out.println("C 1: " + circle1);
        System.out.println("C 2: " + circle2);
        System.out.println("C 3: " + circle3);

        System.out.println();

        // Rectangles
        Rectangle rectangle1 = new Rectangle();
        Rectangle rectangle2 = new Rectangle(new Point(0,0), 5, 10);
        Rectangle rectangle3 = Rectangle.fromArea(new Point(0,0), 50, 5);

        System.out.println("R 1: " + rectangle1);
        System.out.println("R 2: " + rectangle2);
        System.out.println("R 3: " + rectangle3);

        System.out.println();

        // Squares
        Square square1 = new Square();
        Square square2 = new Square(new Point(0,0), 5);
        Square square3 = Square.fromArea(new Point(0,0), 25);

        System.out.println("S 1: " + square1);
        System.out.println("S 2: " + square2);
        System.out.println("S 3: " + square3);

        System.out.println();

        // FormFactory
        FormFactory formFactory = new FormFactory();
        Shape[] shapes = formFactory.produce(10);

        for (Shape shape : shapes) {
            System.out.println(shape);
        }
    }
}

File: ha1/src/main/kotlin/Main.kt
package main.kotlin

/**
 * Main class to demonstrate the functionality of the Circle, Rectangle, and Square classes.
 *
 * @author Simon Wessel
 * @version 1.0
 * @since 1.0
 */
object Main {

    /**
     * Main method to demonstrate the functionality of the classes.
     *
     * @param args Command line arguments (not used).
     */
    @JvmStatic
    fun main(args: Array<String>) {

        println("--- Shapes with kotlin ---")

        // Circles
        val circle1 = Circle()
        val circle2 = Circle(Point(0f, 0f), 5f)
        val circle3 = Circle.fromArea(Point(0f, 0f), 50f)

        println("C 1: $circle1")
        println("C 2: $circle2")
        println("C 3: $circle3")

        println()

        // Rectangles
        val rectangle1 = Rectangle()
        val rectangle2 = Rectangle(Point(0f, 0f), 5f, 10f)
        val rectangle3 = Rectangle.fromArea(Point(0f, 0f), 50f, 5f)

        println("R 1: $rectangle1")
        println("R 2: $rectangle2")
        println("R 3: $rectangle3")

        println()

        // Squares
        val square1 = Square()
        val square2 = Square(Point(0f, 0f), 5f)
        val square3 = Square.fromArea(Point(0f, 0f), 25f)

        println("S 1: $square1")
        println("S 2: $square2")
        println("S 3: $square3")

        println()

        // FormFactory
        val formFactory = FormFactory()
        val shapes = formFactory.produce(10)

        for (shape in shapes) {
            println(shape)
        }
    }
}
